src/luger.erl

```diff
@@ -144,14 +145,46 @@ init_sink(State = #state{}, #syslog_udp_config{ host = Host,
                 syslog_udp_facility = Facility}.
 
 log(Priority, Channel, Message) ->
-    case whereis(?PROC_NAME) of
-        undefined -> {error, luger_not_running};
-        _ -> do_log(Priority, Channel, Message)
-    end.
+    log(Priority, Channel, Message, whereis(?PROC_NAME)).
 
-do_log(Priority, Channel, Message) ->
+log(_Priority, _Channel, _Message, undefined) ->
+    {error, luger_not_running};
+log(Priority, Channel, Message, _) ->
     [{state, State}] = ets:lookup(?TABLE, state),
 
+    Now = erlang:monotonic_time(seconds),
+    Throttle = throttle_channel(Channel, Now, ets:lookup(?TABLE, Channel)),
```

Comment:
That is a little baroque and since this isn't too performance critical, I'll stick with the readable version for now :)